api/request_source/request_source_plugin.proto

```diff
@@ -7,30 +7,47 @@ import "google/protobuf/wrappers.proto";
+// "nighthawk.file-options-list-request-source-plugin")
 // The factory will load the RequestOptionsList from the file, and then passes it to the
 // requestSource it generates. The resulting request source will loop over the RequestOptionsList it
 // is passed.
-message FileBasedPluginConfig {
+message FileBasedOptionsListRequestSourceConfig {
   // The file_path is the path to a file that contains a RequestOptionList in json or yaml format.
+  // This field is required.
   string file_path = 1;
   // The pluginfactory makes requestSources that will generate requests from the RequestOptionList
   // up to num_requests number of times. If num_requests exceeds the number of RequestOptions in the
-  // RequestOptionList located in the file at file_path, it will loop. num_requests = 0 means no
-  // limit on the number of requests to be produced.
-  google.protobuf.UInt32Value num_requests = 2 [(validate.rules).uint32 = {gte: 0, lte: 1000000}];
+  // RequestOptionList located in the file at file_path, it will loop. num_requests = 0 means it
+  // will loop indefinitely, though it will still terminate by normal mechanisms. This field is
+  // optional with a default of 0.
+  uint32 num_requests = 2;
   // The pluginfactory will load the file located in file_path as long as it is below max_file_size,
-  // if it's too large it will throw an error.
+  // if it's too large it will throw an error. This field is optional with a default of 1000000.
   google.protobuf.UInt32Value max_file_size = 3 [(validate.rules).uint32 = {lte: 1000000}];
 }
 
+// Configuration for OptionsListFromProtoRequestSourceFactory (plugin name:
+// "nighthawk.in-line-options-list-request-source-plugin")
+// The resulting request source will loop over the RequestOptionsList it
+// is passed.
+message InLineOptionsListRequestSourceConfig {
+  // The options_list will be used to generate Requests in the RequestSource. This field is
+  // required.
+  nighthawk.client.RequestOptionsList options_list = 1;
+  // The pluginfactory makes requestSources that will generate requests from the RequestOptionList
+  // up to num_requests number of times. If num_requests exceeds the number of RequestOptions in the
+  // options_list, it will loop. num_requests = 0 means it will loop indefinitely, though it will
+  // still terminate by normal mechanisms. This field is optional with a default of 0.
+  uint32 num_requests = 2;
+}
+
 // Configuration for StubPluginRequestSource (plugin name: "nighthawk.stub-request-source-plugin")
 // The plugin does nothing. This is for testing and comparison of the Request Source Plugin Factory
 // mechanism using a minimal version of plugin that does not require a more complicated proto or
 // file reading.
 message StubPluginConfig {
   // test input value which is the only output value in the headers produced from the
-  // requestGenerator for the StubRequestSource.
+  // requestGenerator for the StubRequestSource. This field is optional with a default of 0.
```

Comment:
(nit) Consider removing the comment addition here, since it falls into the a comment category that just "describes the language". It is obvious from the definition below that the field is optional and what its default value is.

The style guide gives some advice on how to write good comments:
https://google.github.io/styleguide/cppguide.html#Comments

A good advice to follow is to avoid writing comments about obvious things, or comments that just describe the language syntax. We should assume that the reader of the comments is familiar with the syntax and just provide comments on tricky parts or logic.

Reply:
Done.